src/Entities/Player.cpp

```diff
@@ -1000,10 +1000,6 @@ void cPlayer::ApplyArmorDamage(int a_DamageBlocked)
 
 bool cPlayer::DoTakeDamage(TakeDamageInfo & a_TDI)
 {
-	SetSpeed(0, 0, 0);
-	// Prevents knocking the player in the wrong direction due to
-	// the speed vector problems, see #2865
-	// In the future, the speed vector should be fixed
```

Comment:
The issue described in this comment is now back.

Reply:
Is resetting the speed when knockback != 0 an acceptable solution?

Reply:
Honestly, I'm not sure.  You'll just have to test it out.